TelosLabs/rails-world
# Feature | Notifications mechanism

## Description
- Implement a reminder system to notify users at specific intervals before an event begins. Initially, the reminders will be set for 15 minutes, 10 minutes, and at the event's start time. 
- [Noticed gem](https://github.com/excid3/noticed) setup.

- [Task](https://github.com/orgs/TelosLabs/projects/5/views/1?pane=issue&itemId=68736380)
## How has this been tested?

Please mark the tests that you ran to verify your changes. If difficult to test, consider providing instructions so reviewers can test.


## Checklist

- [x] CI pipeline is passing
- [x] My code follows the conventions of this project
- [ ] I have added seed data to the database (if applicable)

## How to test

1. Update the REMINDER_CANDENCE constant as below:
```
  REMINDER_CADENCE = {
    first_reminder: 2.minutes,
    second_reminder: 1.minute,
    last_reminder: 0.minutes
  }.
```
2. Ensure `User.last.profile.mail_notifications = true`
3. Run the code below to create some events for testing :
```
(1..6).each_with_index do |cadence, index|
  starts_at = Time.zone.now + cadence.minutes

  event = Event.create!(
    conference: Conference.last,
    title: "Event #{index + 1}",
    description: Faker::Lorem.paragraph,
    starts_at: starts_at,
    ends_at: (starts_at + 10.minutes),
    location: Location.last
  )

  event.users << User.last
end
```
4. The email notifications for each event should be delivered like the table below

<img width="939" alt="image" src="https://github.com/user-attachments/assets/627ffe29-5c34-448f-a40a-efadd9a571a6">
